query/src/pipelines/new/processors/transforms/aggregator/aggregator_final.rs

```diff
@@ -128,6 +130,12 @@ impl<Method: HashMethod + PolymorphicKeysHelper<Method> + Send> Aggregator
         let keys_column = block.column(aggregate_function_len);
         let keys_iter = self.method.keys_iter_from_column(keys_column)?;
 
+        let group_by_two_level_threshold =
+            self.ctx.get_settings().get_group_by_two_level_threshold()? as usize;
+        if !self.state.is_two_level() && block.num_rows() >= group_by_two_level_threshold {
```

Comment:
maybe we should use `map.size` to replace block.num_rows ?

Reply:
make sense. I will test it.

Reply:
If change to `self.state.len()`, I also need to change `convert_to_two_level` to copy the old map. Because I create a new empty `two_level_hash_table` now. I will change it.